arbnode/batch_poster.go

```diff
@@ -493,9 +493,8 @@ func (b *BatchPoster) addEspressoBlockMerkleProof(
 		if err != nil {
 			return fmt.Errorf("could not get the merkle root at height %v", jst.Header.Height)
 		}
-		if snapshot.Height != b.currentValidatedHotShotHeight {
```

Comment:
The snapshot height MUST be greater than the height of the leaf we are proving, because the leaf we are proving is only a valid leaf of the merkle tree in the following blocks.

Reply:
I can't understand. The `snapshop.Height` is fetched thourgh the `jst.Header.Height` and I think it already indicates that the `snapshot.Height` is greater than `jst.Header.Height`. Otherwise an error should be returned